src/core/components/ChatInput.tsx

```diff
@@ -131,7 +142,11 @@ const ChatInput = ({
             aria-label="Send Chat"
             disabled={
               (!isRequesting && inputValue.length === 0) ||
-              !isWithinTokenLimit(inputValue, modelConfig.contextLength)
+              !hasSufficientRemainingTokens(
+                modelConfig.id,
+                inputValue,
+                remainingContextWindow,
+              )
```

Comment:
- For gpt, we have a 3rd party hook that checks 
- For deepseek, we have to do a rough estimate by characters since we only know token usage _after_ the model responds